test/e2e/network_segmentation_services.go

```diff
@@ -133,18 +117,28 @@ var _ = Describe("Network Segmentation: services", func() {
 							TargetPort: intstr.FromInt(serviceTargetPort),
 						},
 					}
-					s.Spec.Type = v1.ServiceTypeNodePort
+					s.Spec.Type = v1.ServiceTypeLoadBalancer
 					s.Spec.IPFamilyPolicy = &policy
 				})
 				framework.ExpectNoError(err)
 
+				By("Wait for UDN LoadBalancer Ingress to pop up")
+				udnService, err = jig.WaitForLoadBalancer(context.TODO(), 180*time.Second)
+				framework.ExpectNoError(err)
+
 				By("Creating a UDN backend pod")
 				udnServerPod := e2epod.NewAgnhostPod(
 					namespace, "backend-pod", nil, nil,
 					[]v1.ContainerPort{
 						{ContainerPort: (serviceTargetPort), Protocol: "UDP"}},
-					"netexec",
-					"--udp-port="+fmt.Sprint(serviceTargetPort))
+					"-c",
+					fmt.Sprintf(`
+set -xe
+iface=ovn-udn1
+ips=$(ip -o addr show dev $iface| grep global |awk '{print $4}' | cut -d/ -f1 | paste -sd, -)
```

Comment:
nit: add a comment here to indicate you are parsing the IPs on udn1 interface within the pod and then using that as the listener IPs? Is this really needed? why not just listen on all IPs?

Reply:
To be sure that we are not answering back over the wrong interface if we break ovn-k, so more or less to reduce the possibility of a false possitive.